sklearn/tree/tree.py

```diff
@@ -297,9 +297,13 @@ def fit(self, X, y, sample_weight=None, check_input=True,
                 sample_weight = expanded_class_weight
 
         # Set min_weight_leaf from min_weight_fraction_leaf
-        if self.min_weight_fraction_leaf != 0. and sample_weight is not None:
+        if self.min_weight_fraction_leaf != 0.:
+            if sample_weight is None:
+                sample_weight = np.repeat(1., n_samples)
```

Comment:
If `sample_weight is None` then `min_weight_fraction_leaf` has no effect. I think that's quite clear from the `else` case here.